### PR TITLE
Add dogstatsd_entity_id_precedence parameter

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Add new dogstatsd server parameter to activate or not a new feature.
During QA we discovered that the new `DD_ENTITY_ID` tagging precedence in dogstatsd should be behind a configuration variable and defaulted to "false" by default. 

### Motivation

In a specific configuration: UDS and ENTITY_ID dogstatsd origin detection is activated, but the ENTITY_ID is not valid, the tags coming from the UDS detection are not added.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -293,6 +293,7 @@ func initConfig(config Config) {
 	config.BindEnvAndSetDefault("dogstatsd_tags", []string{})
 	config.BindEnvAndSetDefault("dogstatsd_mapper_cache_size", 1000)
 	config.BindEnvAndSetDefault("dogstatsd_string_interner_size", 4096)
+	config.BindEnvAndSetDefault("dogstatsd_entity_id_precedence", false)
```
Comment (xornivore) [inline pkg/config/config.go]: Might be worth adding a comment here or document the related boolean field so it's clear what this new setting accomplishes?

**Review by xornivore — DISMISSED** [review]
LGTM just a minor comment about documenting it

File: pkg/config/config.go (hunk 1 above)
Comment (xornivore) [inline pkg/config/config.go]:
```suggestion
	// Enable check for Entity-ID presence when enriching Dogstatsd metrics with tags
	config.BindEnvAndSetDefault("dogstatsd_entity_id_precedence", false)
```

**Review by xornivore — DISMISSED** [review]
👍

File: pkg/config/config.go (hunk 1 above)
Comment (hush-hush) [reply pkg/config/config.go]: Do we need to add this to the datadog.yaml.example ? When will this be set to true ?
If users are meant to use this flag we should document it.

Also if we have a release note describing the new feature related to `dogstatsd_entity_id_precedence` we should edit it to maybe explain this flag to ?

Comment (xornivore) [reply pkg/config/config.go]: Added a release note.

**Review by sarina-dd — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Comment by olivielpeau** [comment]
A couple of questions I think we should follow up on after 7.18, through some docs at least:

* long-term, how can a user know whether they can enable `dogstatsd_entity_id_precedence`?
* what are the benefits/risks of enabling it?

**Comment by clamoriniere** [comment]
> * long-term, how can a user know whether they can enable `dogstatsd_entity_id_precedence`?
> * what are the benefits/risks of enabling it?

@olivielpeau 
this option should be activated if the user suffers of having tagging conflict when they use UDS, or they don't want to add the origin tags.

**Comment by cptkng** [comment]
Could someone point me to the documentation for this feature on the DD website?!

**Pull request merged**